crates/runtime/src/dataaccelerator.rs

```diff
@@ -439,6 +439,25 @@ impl AcceleratorExternalTableBuilder {
     }
 }
 
+/// Represents acceleration source component, such as a dataset or a view.
+/// Provides additional information about the source, such as its name and associated runtime information.
+pub trait AccelerationSource: Sync {
```

Comment:
Alternatively this can be a struct/wrapper on top of dataset/view to provide unified functionality